src/test/java/teammates/ui/webapi/UpdateNotificationActionTest.java

```diff
@@ -0,0 +1,132 @@
+package teammates.ui.webapi;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.NotificationStyle;
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.util.Const;
+import teammates.ui.output.NotificationData;
+import teammates.ui.request.InvalidHttpRequestBodyException;
+import teammates.ui.request.NotificationUpdateRequest;
+
+/**
+ * SUT: {@link UpdateNotificationAction}.
+ */
+public class UpdateNotificationActionTest extends BaseActionTest<UpdateNotificationAction> {
+    @Override
+    String getActionUri() {
+        return Const.ResourceURIs.NOTIFICATION;
+    }
+
+    @Override
+    String getRequestMethod() {
+        return PUT;
+    }
+
+    @Test
+    @Override
+    protected void testExecute() throws Exception {
+        final String TEST_NOTIFICATION = "notification1";
+        NotificationAttributes testNotificationAttribute = typicalBundle.notifications.get(TEST_NOTIFICATION);
+
+        String[] requestParams = new String[] {
+                Const.ParamsNames.NOTIFICATION_ID, testNotificationAttribute.getNotificationId(),
+        };
+        NotificationUpdateRequest req = getTypicalUpdateRequest();
+        NotificationStyle style = req.getStyle();
+        NotificationTargetUser targetUser = req.getTargetUser();
+        String title = req.getTitle();
+        String message = req.getMessage();
+        String invalidTitle = "";
+        String invalidNotificationId = "InvalidNotificationId";
+
+        loginAsAdmin();
+
+        ______TS("Typical Case: Update notification successfully");
+        req = getTypicalUpdateRequest();
+        UpdateNotificationAction action = getAction(req, requestParams);
+        NotificationData res = (NotificationData) action.execute().getOutput();
+
+        NotificationAttributes updatedNotification = logic.getNotification(res.getNotificationId());
+
+        // Verify that correctly updated in the DB
+        assertEquals(req.getStartTimestamp(), updatedNotification.getStartTime().toEpochMilli());
+        assertEquals(req.getEndTimestamp(), updatedNotification.getEndTime().toEpochMilli());
+        assertEquals(style, updatedNotification.getStyle());
+        assertEquals(targetUser, updatedNotification.getTargetUser());
+        assertEquals(title, updatedNotification.getTitle());
+        assertEquals(message, updatedNotification.getMessage());
+
+        ______TS("Parameters cannot be null");
+        req = getTypicalUpdateRequest();
```

Comment:
Sorry for not picking this up this earlier, actually I think we don't need to test null params here because the validation by `getAndValidateRequestBody` will guarantee they are non-null.

Reply:
I see, but should we still keep this in for black box testing?

Reply:
Technically there's no harm having those tests, let's keep it first then. Although I certainly think that they are not the most important tests to have.

Reply:
Sure, one other reason is that the `UpdateNotificationRequest` is not tested, doesn't look like requests themselves are tested directly, referencing the others